hub-templates/basehub/values.yaml

```diff
@@ -287,3 +288,28 @@ jupyterhub:
             return resp
 
         c.JupyterHub.authenticator_class = CustomOAuthenticator
+
+      07-cloud-storage-bucket: |
+        from z2jh import get_config
+        cloud_resources = get_config('custom.cloudResources')
+        scratch_bucket = cloud_resources['scratchBucket']
+        import os
+
+        if scratch_bucket['enabled']:
+          # FIXME: Support other providers too
+          assert cloud_resources['provider'] == 'gcp'
+          project_id = cloud_resources['gcp']['projectId']
+
+          release = os.environ['HELM_RELEASE_NAME']
+          bucket_protocol = 'gcs'
+          bucket_name = f'{project_id}-{release}-scratch-bucket'
+          env = {
+            'SCRATCH_BUCKET_PROTOCOL': bucket_protocol,
+            # Matches "daskhub.scratchBUcket.name" helm template
+            'SCRATCH_BUCKET_NAME': bucket_name,
+            # Use k8s syntax of $(ENV_VAR) to substitute env vars dynamically in other env vars
+            'SCRATCH_BUCKET': f'{bucket_protocol}://{bucket_name}/$(JUPYTERHUB_USER)',
+            'PANGEO_SCRATCH': f'{bucket_protocol}://{bucket_name}/$(JUPYTERHUB_USER)',
+          }
+
+          c.KubeSpawner.environment.update(env)
```

Comment:
@yuvipanda, I had the impression this piece was daskhub-specific? Why did you moved it to the basehub?

Reply:
cloudResoures was already under `jupyterhub.custom`, rather `daskhub.custom` - `get_config` can only read things under `jupyterhub`. None of this is specific to dask-gateway either - you can just as well want an s3 bucket without dask-gateway. Finally, for https://github.com/2i2c-org/pilot-hubs/issues/362 I've been trying to slowly remove everything under daskhub/ that doesn't strictly belong there...

Reply:
> None of this is specific to dask-gateway either - you can just as well want an s3 bucket without dask-gateway. Finally, for #362 I've been trying to slowly remove everything under daskhub/ that doesn't strictly belong there...

@yuvipanda it could be reasonable to let the JupyterHub helm chart stash away `global` values into the k8s Secret that is exposed to the hub pod, but I'm not sure. Open an issue in z2jh if you want to suggest such feature is implemented.

Reply:
I want to remove the extra chart anyway, for #362. So I think it's ok to keep it like this for now

Reply:
> None of this is specific to dask-gateway either - you can just as well want an s3 bucket without dask-gateway.

Gotcha! I had the wrong impression because I heard about the scratch bucket in a Dask context, but you are actually right about the scratch bucket could be also useful in other environments.
Thanks for the clarification!